crates/bevy_pbr/src/decal/forward.rs

```diff
@@ -146,3 +148,16 @@ impl Default for ForwardDecalMaterialExt {
         }
     }
 }
+
+#[derive(Resource)]
+struct ForwardDecalMesh(Handle<Mesh>);
+
+fn forward_decal_set_mesh(mut world: DeferredWorld, HookContext { entity, .. }: HookContext) {
```

Comment:
This needs a comment about why we can't just use required components. I think I agree with your reasoning, but it needs to be recorded in the code base for future contributors to evaluate quickly if it's still valid.

Reply:
Done! It's quite a long comment haha